config/main.tf

```diff
@@ -0,0 +1,39 @@
+terraform {
+  backend "s3" {
+    key = "terraform/build/callisto-auth-keycloak"
+  }
+}
+
+provider "keycloak" {
+  # Variables in here must be defined locally before running the plan
+  client_id     = var.tf_client_id
+  client_secret = var.tf_client_secret
+  realm         = var.tf_realm
+  url           = var.tf_url
+}
+
+data "keycloak_realm" "callisto_notprod_auth" {
+  realm = var.tf_realm
+}
+
+
+resource "keycloak_openid_client" "callisto_ui" {
```

Comment:
the resource name and the access_type being public suggested this was the client for the UI. However it has a client id of terraform-client which suggest maybe it is for the client that will be used to run the GitHub action. However that wouldn't have a public access_type. Also in conversation you mentioned you are still working on the client for the UI which further confuses me about what this resource is for

Reply:
I've removed this as it was just for testing that everything was working on my end